lib/src/async_cache.dart

```diff
@@ -62,8 +74,27 @@ class AsyncCache<T> {
     if (_cachedStreamSplitter != null) {
       throw StateError('Previously used to cache via `fetchStream`');
     }
-    return _cachedValueFuture ??= callback()
-      ..whenComplete(_startStaleTimer).ignore();
+    if (_canCacheException) {
+      return _cachedValueFuture ??= callback()
+        ..whenComplete(_startStaleTimer).ignore();
+    } else {
+      if (_cachedValueFuture == null) {
+        try {
+          ///First we run the callback then we assign the value received
+          ///from [callback] to the [_cachedValueFuture]
+          var value = await callback();
```

Comment:
Rite, we were waiting for the callback to complete then, we were assigning the value to the cache. While the callback is getting executed, we were able to execute callback once again because value inside the cache was not updated.

Reply:
We cannot use rethrow because it requires a catch clause